tests/e2e/main.go

```diff
@@ -97,57 +140,147 @@ func executeTests(tests []testRunWithSteps) (err error) {
 	return
 }
 
+func getTestCaseUsageString() string {
+	var builder strings.Builder
+
+	// Test case selection
+	builder.WriteString("This flag is used to reference existing, defined test cases to be run.")
+	builder.WriteString("Test case selection:\nSelection of test steps to be executed:\n")
+	for _, stepChoice := range stepChoices {
+		builder.WriteString(fmt.Sprintf("- %s : %s. Compatible with test runners: %s\n", stepChoice.name, stepChoice.description, strings.Join(stepChoice.testRuns, ",")))
+	}
+	builder.WriteString("\n")
+
+	// Test runner selection
+	builder.WriteString("Test runner selection:\nSelection of test runners to be executed:\n")
+	for _, testRunChoice := range testRuns {
+		builder.WriteString(fmt.Sprintf("- %s : %s\n", testRunChoice.name, testRunChoice.description))
+	}
+	builder.WriteString("\n")
+
+	// Example
+	builder.WriteString("Example: -tc multiconsumer::multiconsumer -tc happy-path::default")
+
+	return builder.String()
+}
+
+func getTestFileUsageString() string {
+	var builder strings.Builder
+
+	builder.WriteString("This flag is used to reference files containing step traces to be run.\n")
+	builder.WriteString("Each filename should be separated by '::' from the test runner name.\n")
+
+	// Test runner selection
+	builder.WriteString("Test runner selection:\nSelection of test runners to be executed:\n")
+	for _, testRunChoice := range testRuns {
+		builder.WriteString(fmt.Sprintf("- %s : %s\n", testRunChoice.name, testRunChoice.description))
+	}
+	builder.WriteString("\n")
+
+	// Example
+	builder.WriteString("Example: -test-file awesome-trace.json::default -test-file other-trace.json::default")
+
+	return builder.String()
+}
+
 func parseArguments() (err error) {
-	flag.Var(&testSelection, "tc",
-		fmt.Sprintf("Selection of test cases to be executed:\n%s,\n%s",
-			func() string {
-				var keys []string
-				for k, v := range testMap {
-					keys = append(keys, fmt.Sprintf("- %s : %s", k, v.description))
-				}
-				return strings.Join(keys, "\n")
-			}(),
-			"Example: -tc multiconsumer -tc happy-path "))
+	flag.Var(&selectedTests, "tc",
+		getTestCaseUsageString())
+
+	flag.Var(&selectedTestfiles, "test-file",
+		getTestFileUsageString())
 	flag.Parse()
 
 	// Enforce go-relayer in case of cometmock as hermes is not yet supported
 	if useCometmock != nil && *useCometmock && (useGorelayer == nil || !*useGorelayer) {
 		fmt.Println("Enforcing go-relayer as cometmock is requested")
 		if err = flag.Set("use-gorelayer", "true"); err != nil {
-			return
-		}
-	}
-	// check if specified test case exists
-	for _, tc := range testSelection {
-		if _, hasKey := testMap[tc]; !hasKey {
-			err := fmt.Errorf("unknown test case '%s'", tc)
 			return err
 		}
 	}
-	return
+	return nil
 }
 
-func getTestCases(selection TestSet) (tests []testRunWithSteps) {
+type testRunWithSteps struct {
+	testRun TestRun
+	steps   []Step
```

Comment:
Related to above, it seems like "steps" should be a field of `TestRun`, as in a test run is a high level abstraction with config + actual steps to be executed

Reply:
I have renamed TestRun to TestConfig, since that is much more fitting to reality.

Good call, easy to get blind to these weird, historical naming choices. I've refactored this.

The abstractions are now:
A TestRun is made of 
steps []Step
testConfig TestConfig

lmk if you still find it confusing